File: frontend/src/types/device.ts
import { Customer } from "./customer";
import { DeviceModel } from "./deviceModel";

export interface Device {
  id: number;
  customer: Customer;
  model: DeviceModel;
  serialNumber: string;
  purchaseDate: string | null;
  installationDate: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface CreateDeviceRequest {
  customer: { id: number };
  model: { id: number };
  serialNumber: string;
  purchaseDate?: string;
  installationDate?: string;
}

export interface UpdateDeviceRequest {
  customer: { id: number };
  model: { id: number };
  serialNumber: string;
  purchaseDate?: string;
  installationDate?: string;
}

File: backend/src/main/java/com/servis/backend/entity/Device.java
package com.servis.backend.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "devices")
public class Device {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotNull(message = "Müşteri bilgisi zorunludur")
    @ManyToOne
    @JoinColumn(name = "customer_id", nullable = false)
    private Customer customer;

    @NotNull(message = "Model bilgisi zorunludur")
    @ManyToOne
    @JoinColumn(name = "model_id", nullable = false)
    private DeviceModel model;

    @NotBlank(message = "Seri numarası boş olamaz")
    @Size(max = 150, message = "Seri numarası en fazla 150 karakter olabilir")
    @Column(name = "serial_number", length = 150, unique = true, nullable = false)
    private String serialNumber;

    @Column(name = "purchase_date")
    private LocalDate purchaseDate;

    @Column(name = "installation_date")
    private LocalDate installationDate;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
    
    
    
}